apps/zotonic_core/src/models/m_dispatch.erl

```diff
@@ -0,0 +1,109 @@
+%% @author Marc Worrell <marc@worrell.nl>
+%% @copyright 2025 Marc Worrell
+%% @doc Model for dispatching requests.
+%% @end
+
+%% Copyright 2025 Marc Worrell
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+
+-module(m_dispatch).
+-author("Marc Worrell <marc@worrell.nl").
+
+-behaviour(zotonic_model).
+
+-export([
+    m_get/3,
+
+    dispatch_url/2,
+    dispatch_path/2
+]).
+
+m_get([ <<"url">> ], #{ payload := #{ <<"url">> := Url } }, Context) ->
+    dispatch_url(Url, Context);
+m_get([ <<"url">> ], _Msg, Context) ->
+    dispatch_url(z_context:get_q(<<"url">>, Context), Context);
+m_get([ <<"url">>, Url ], _Msg, Context) ->
+    dispatch_url(Url, Context);
+m_get([ <<"path">> ], #{ payload := #{ <<"path">> := Path } }, Context) ->
+    dispatch_path(Path, Context);
+m_get([ <<"path">> ], _Msg, Context) ->
+    dispatch_path(z_context:get_q(<<"path">>, Context), Context);
+m_get([ <<"path">> | Path ], _Msg, Context) ->
+    dispatch_path(Path, Context);
+m_get(_, _Msg, _Context) ->
+    {error, enoent}.
```

Comment:
What do you think of adding another path here to instead access the result of `z_dispatcher:url_for`?

I think that would also be useful for external services/apps.

Reply:
To generate URLs?  That is a good idea.